src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,185 @@
+                    LocalDateTime startDateTime = null;
+                    String startDate = null;
+
+                    // Try parsing with different formats for starting date and time input.
+                    for (String formatString : formatStrings) {
+                        try {
+                            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(formatString);
+                            startDateTime = LocalDateTime.parse(startDateInput, formatter);
+                            startDate = startDateTime.format(formatter);
+                            break;  // Stop at the first successful parse
+                        } catch (DateTimeParseException e) {
+                            // Ignore the exception and try the next format
+                        }
+                    }
+
+                    // if the input fits none of the format, throw DateTimeParseException
+                    if (startDate == null) {
+                        throw new DateTimeParseException("Invalid date/time format", startDate, 0);
+                    }
+                    LocalDateTime dueDateTime = null;
+                    String dueDate = null;
+
+                    // Try parsing with different formats for due date and time input.
+                    for (String formatString : formatStrings) {
+                        try {
+                            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(formatString);
+                            dueDateTime = LocalDateTime.parse(dueDateInput, formatter);
+                            dueDate = dueDateTime.format(formatter);
+                            break;  // Stop at the first successful parse
+                        } catch (DateTimeParseException e) {
+                            // Ignore the exception and try the next format
+                        }
+                    }
+
+                    // if the input fits none of the format, throw DateTimeParseException
+                    if (dueDate == null) {
+                        throw new DateTimeParseException("Invalid date/time format", dueDateInput, 0);
+                    }
+
+                    // check if start date is before due date.
+                    if (startDateTime.isAfter(dueDateTime) || startDateTime.isEqual(dueDateTime)) {
+                        throw new IllegalArgumentException();
+                    }
+
+                    // add item into list
+                    Event task = new Event(taskName, startDate, dueDate);
+                    return new AddCommand(task);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    if (input.split(" ").length == 1) {
+                        return new ErrorCommand("☹ OOPS!!! The description of a event cannot be empty");
+                    } else {
+                        return new ErrorCommand("☹ OOPS!!! The format of starting date or due date is wrong, please write it in the format correct format");
```

Comment:
Perhaps this line is too long? It seems to be exceeding the character limit of 120 per line according to the style guide.
May be you want to consider breaking it up?